CMakeLists.txt

```diff
@@ -1,55 +1,56 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(srdfdom)
 
-find_package(Boost REQUIRED)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
+find_package(Boost REQUIRED)
 find_package(console_bridge REQUIRED)
 find_package(urdfdom_headers REQUIRED)
-
-find_package(catkin REQUIRED COMPONENTS cmake_modules urdf urdfdom_py)
-
-find_package(TinyXML REQUIRED)
-
-include_directories(
-  include
-  ${Boost_INCLUDE_DIR}
-  ${TinyXML_INCLUDE_DIRS}
-  ${catkin_INCLUDE_DIRS}
-  ${console_bridge_INCLUDE_DIRS}
-  ${urdfdom_headers_INCLUDE_DIRS}
-  )
-
-add_compile_options(-std=c++11)
-
-catkin_python_setup()
-
-catkin_package(
-  LIBRARIES ${PROJECT_NAME}
-  INCLUDE_DIRS include
-  DEPENDS TinyXML console_bridge urdfdom_headers
-)
-
-add_library(${PROJECT_NAME} 
+find_package(ament_cmake REQUIRED)
+find_package(urdf REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
+
+include_directories(include ${rclcpp_INCLUDE_DIRS}
+                    ${rmw_implementation_INCLUDE_DIRS}
+                    ${Boost_INCLUDE_DIR}
+                    ${TinyXML_INCLUDE_DIRS}
+                    ${console_bridge_INCLUDE_DIRS}
+                    ${urdfdom_headers_INCLUDE_DIRS}
+                    ${std_msgs_INCLUDE_DIRS})
+
+add_library(${PROJECT_NAME}
   src/model.cpp
   src/srdf_writer.cpp
 )
-target_link_libraries(${PROJECT_NAME} ${TinyXML_LIBRARIES} ${catkin_LIBRARIES} ${console_bridge_LIBRARIES} ${urdfdom_headers_LIBRARIES})
-
+ament_target_dependencies(${PROJECT_NAME}
+        ${TinyXML_LIBRARIES}
+        ${console_bridge_LIBRARIES}
+        urdfdom_headers
+        urdf)
 
 install(TARGETS ${PROJECT_NAME}
-  DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  DESTINATION lib
 )
 install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+  DESTINATION include/${PROJECT_NAME}
   FILES_MATCHING PATTERN "*.h"
 )
 
-install(PROGRAMS 
+install(PROGRAMS
   scripts/display_srdf
-  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  DESTINATION lib/${PROJECT_NAME}
 )
 
-if(CATKIN_ENABLE_TESTING)
-  find_package(rostest REQUIRED)
-  add_rostest(test/srdf_parser.test)
-endif()
+# if(BUILD_TESTING)
+#   find_package(rostest REQUIRED)
+#   add_rostest(test/srdf_parser.test)
+# endif()
```

Comment:
we should never disable tests, the likely hood that someone comes back and re-enables this is low due to human nature. tests are our friends